aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -162,8 +168,18 @@ def _get_hashed_idempotency_key(self, lambda_event: Dict[str, Any]) -> str:
         data = lambda_event
         if self.event_key_jmespath:
             data = self.event_key_compiled_jmespath.search(lambda_event)
+
+        if self.is_missing_idempotency_key(data):
+            warnings.warn(f"No value found for idempotency_key. jmespath: {self.event_key_jmespath}")
+            if self.raise_on_no_idempotency_key:
+                raise IdempotencyKeyError("No data found to create a hashed idempotency_key")
+
         return self._generate_hash(data)
 
+    @staticmethod
+    def is_missing_idempotency_key(data) -> bool:
+        return data is None or (type(data) is list and all(x is None for x in data))
```

Comment:
```suggestion
        return data is None or not data or all(x is None for x in data)
```

This accounts for empty data structures, including non-lists iterables like tuples, dictionaries